src/Core/Extractors/nNASOS.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace ArchiveCacheManager
+{
+    // this extractor is specificilly for nNASOS1.8 available at https://archive.org/details/nNASOS1.8
+    public class NNASOS : Extractor
+    {
+        // Needed to place in LanchBox root ThirdParty/ folder as nNASOS uses a .dll itself that LaunchBox
+        // mistakenly tries to register as a plugin, crashing LaunchBox on startup.
+        string executablePath = Path.Combine(PathUtils.GetLaunchBoxNNASOSRootPath(), "nNASOS.exe");
+
+        public NNASOS()
+        {
+
+        }
+
+        public static bool SupportedType(string archivePath)
+        {
+            return PathUtils.HasExtension(archivePath, new string[] { ".7z", ".zip", ".iso.dec" });
+        }
+
+        public override bool Extract(string archivePath, string cachePath, string[] includeList = null, string[] excludeList = null)
+        {
+            string isoDecPath = string.Empty;
+            string extension = Path.GetExtension(archivePath).ToLower();
+            bool skipArchiveCheck = false;
+
+            if (Directory.Exists(cachePath))
+            {
+                string[] files = Directory.GetFiles(cachePath, "*.iso.dec");
+                if (files.Length > 0)
+                {
+                    // .iso.dec is already in the cache path, so no need to extract or copy anything
+                    // from archive path
+                    Logger.Log(string.Format("Skipping extract/copy from archive path as .iso.dec is already in cache path..."));
+                    isoDecPath = files[0];
+                    skipArchiveCheck = true;
+                }
+            }
+
+            if (!skipArchiveCheck)
+            {
+                if (extension.Equals(".7z") || extension.Equals(".zip"))
+                {
+                    Extractor zip = new Zip();
+                    if (zip.Extract(archivePath, cachePath, new string[] { "*.iso", "*.iso.dec" }, excludeList))
+                    {
+                        string[] files = Directory.GetFiles(cachePath, "*.iso");
+                        if (files.Length > 0)
+                        {
+                            // no need to further process with nNASOS as already in .iso format
+                            return true;
+                        }
+                        else
+                        {
+                            isoDecPath = Directory.GetFiles(cachePath, "*.iso.dec")[0];
+                        }
+                    }
+                }
+                else if (extension.Equals(".iso.dec"))
```

Comment:
`Path.GetExtension()` will return .dec for .iso.dec files
```suggestion
                else if (extension.Equals(".dec"))
```